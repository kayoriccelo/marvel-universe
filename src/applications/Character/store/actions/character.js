import api from '../../../../services/api'
import * as Types from '../types'
import history from '../../../../history'


export const getListingCharactersAPI = params => {
    let search = params['search'] ? `?nameStartsWith=${params['search']}` : ''

    return dispatch => {
        return api.get(`/v1/public/characters${search}`).then(response => {
            dispatch({ type: Types.LISTING_CHARACTERS, payload: response.data.data.results })

            return response.data.results
        })
    }
}

export const getListingCharacters = _ => {
    return dispatch => dispatch({ type: Types.LISTING_CHARACTERS })
}

export const getListingSeries = character => {
    return dispatch => {
        return api.get(`/v1/public/characters/${character}/series`).then(response => {
            let series = response.data.data ? response.data.data.results : response.data

            dispatch({ type: Types.LISTING_SERIES, payload: series })

            return series
        })
    }
}

export const clearSeries = _ => {
    return dispatch => {
        dispatch({ type: Types.LISTING_SERIES, payload: [] })
    }
}

export const loadCharacter = params => {
    return dispatch => {
        return api.get(`/v1/public/characters/${params['id']}`).then(response => {
            if (response.data.data !== undefined) {
                let character = response.data.data.results[0]
                dispatch([{ type: Types.LOAD_CHARACTER, payload: character }, getListingSeries(character.id)])
            } else {
                let character = response.data
                dispatch({ type: Types.LOAD_CHARACTER, payload: character })
            }
        })
    }
}

export const updateCharacter = params => {
    return dispatch => {
        let itens = params['itens'].map(item => item.id === params['instance'].id ? params['instance'] : item)

        dispatch([
            { type: Types.UPDATE_CHARACTER, payload: params['instance'] },
            { type: Types.LISTING_CHARACTERS, payload: itens }
        ])

        history.push('/characters/listing')
    }
}