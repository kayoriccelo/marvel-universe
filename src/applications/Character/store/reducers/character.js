import * as Types from '../types'


export const initialState = {
    itens: [],
    series: [],
    instance: null
}

export const characterReducer = (state = initialState, action) => {
    switch (action.type) {

        case Types.LISTING_CHARACTERS:
            return { ...state, itens: action.payload || state.itens }

        case Types.LOAD_CHARACTER:
            return { ...state, instance: action.payload }

        case Types.UPDATE_CHARACTER:
            return { ...state, instance: action.payload }

        case Types.LISTING_SERIES:
            return { ...state, series: action.payload }

        default:
            return state
    }
}
