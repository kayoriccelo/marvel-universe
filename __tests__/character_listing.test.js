import { shallowListingCharacterSetup } from './character.config'
import sinon from 'sinon'

describe('Shallow rendered Character Listing', () => {
    let wrapper

    beforeEach(() => {
        const { enzymeWrapper } = shallowListingCharacterSetup()
        wrapper = enzymeWrapper
    })

    afterEach(() => {})

    it('should render a list characters', () => {

        expect(wrapper).toMatchSnapshot()

    })
})