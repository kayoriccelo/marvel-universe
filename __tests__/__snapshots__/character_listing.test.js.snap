// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendered Character Listing should render a list characters 1`] = `ShallowWrapper {}`;
